### PR TITLE
JDK-8279956: Useless method Scheduling::ComputeLocalLatenciesForward()

### PR DESCRIPTION
**Pull request by sunny868**
Method `Scheduling::ComputeLocalLatenciesForward()` be used for compute the latency of all the instructions and write the result to `_node_latency`. But I found that `_node_latency` was never used. So I think method `Scheduling::ComputeLocalLatenciesForward()` can be deleted, which is also an enhancement to the efficiency of compilation threads.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8279956](https://bugs.openjdk.java.net/browse/JDK-8279956): Useless method Scheduling::ComputeLocalLatenciesForward()

### Reviewers
 * [Jie Fu](https://openjdk.java.net/census#jiefu) (@DamonFool - **Reviewer**)
 * [Nils Eliasson](https://openjdk.java.net/census#neliasso) (@neliasso - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7062/head:pull/7062` \
`$ git checkout pull/7062`

Update a local copy of the PR: \
`$ git checkout pull/7062` \
`$ git pull https://git.openjdk.java.net/jdk pull/7062/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7062`

View PR using the GUI difftool: \
`$ git pr show -t 7062`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7062.diff">https://git.openjdk.java.net/jdk/pull/7062.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/output.cpp [hunk 1]
```diff
@@ -22,6 +22,12 @@
  *
  */
 
+/*
+ * This file has been modified by Loongson Technology in 2022. These
+ * modifications are Copyright (c) 2022, Loongson Technology, and are made
```
Comment (DamonFool) [inline src/hotspot/share/opto/output.cpp:27]: I'm afraid this part of change is unnecessary.
Maybe, you can just remove it, right?

Comment (sunny868) [reply src/hotspot/share/opto/output.cpp:27]: Thanks for the reminder.

**Comment by DamonFool** [comment]
It seems that you didn't enable Github Actions on your personal forks that offer testing.
So your PRs won't be tested before integration.

Please enable it: https://wiki.openjdk.java.net/display/SKARA/Testing#Testing-Configuringworkflowstorun

**Comment by sunny868** [comment]
> It seems that you didn't enable Github Actions on your personal forks that offer testing. So your PRs won't be tested before integration.
> 
> Please enable it: https://wiki.openjdk.java.net/display/SKARA/Testing#Testing-Configuringworkflowstorun

It seems that you didn't enable Github Actions on your personal forks that offer testing. So your PRs won't be tested before integration.

I had set `Allow all actions` for my personal forks `sunny868/jdk`, but there is a warning `GitHub Actions is currently disabled for your account. ` I do not know why? 
I have turned to ` GitHub Support ` for a help.

**Comment by neliasso** [comment]
Have you done any archaeology to determine if this code was used in the past?

**Comment by sunny868** [comment]
> 
I only follow the following earliest changeset from the `hg.openjdk.java.net/jdk/jdk` library

changeset:   1:489c9b5090e2
user:        duke
date:        Sat Dec 01 00:00:00 2007 +0000
summary:     Initial load

_node_latency alse be unused. I think latency has be done on LCM phase. so here` DoSheduling` is nothing to do for node latency.

**Comment by DamonFool** [comment]
Please clarify what tests you've done.
Thanks.

**Comment by dean-long** [comment]
I checked the SCCS history back to 2000 when _node_latency was introduced, and could not find any use, other than the tracing output.

**Comment by DamonFool** [comment]
This change looks reasonable to me.
But I would suggest testing with -XX:+OptoScheduling before integration.

**Comment by sunny868** [comment]
> 

Thanks. Can I trigger  on the private PR? What if you can do it?

> This change looks reasonable to me. But I would suggest testing with -XX:+OptoScheduling before integration.

Thanks, Can I re-run my workflow with `-XX:+OptoScheduling` on your personal forks? How to do?

**Comment by DamonFool** [comment]
> > 
> 
> Thanks. Can I trigger on the private PR? What if you can do it?
> 
> > This change looks reasonable to me. But I would suggest testing with -XX:+OptoScheduling before integration.
> 
> Thanks, Can I re-run my workflow with `-XX:+OptoScheduling` on your personal forks? How to do?

Maybe, you can just run tier1~3 with `-XX:+OptoScheduling` on your local x86 machine and feedback here.

**Comment by sunny868** [comment]
Test result on x86 used command `make test TEST=":tier1 :tier2 :tier3" CONF=fast JTREG="VM_OPTIONS=-XX:+OptoScheduling;TIMEOUT_FACTOR=10"` 

<pre>
==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
   jtreg:test/hotspot/jtreg:tier1                     1765  1765     0     0   
>> jtreg:test/jdk:tier1                               2072  2071     1     0 <<
   jtreg:test/langtools:tier1                         4243  4243     0     0   
   jtreg:test/jaxp:tier1                                 0     0     0     0   
   jtreg:test/lib-test:tier1                             0     0     0     0   
   jtreg:test/hotspot/jtreg:tier2                      441   441     0     0   
>> jtreg:test/jdk:tier2                               3784  3770    10     4 <<
   jtreg:test/langtools:tier2                           11    11     0     0   
   jtreg:test/jaxp:tier2                               452   452     0     0   
   jtreg:test/hotspot/jtreg:tier3                       72    72     0     0   
   jtreg:test/jdk:tier3                                174   174     0     0   
   jtreg:test/langtools:tier3                            0     0     0     0   
   jtreg:test/jaxp:tier3                                 0     0     0     0   
==============================
TEST FAILURE
</pre>

 The test FAIL and ERROR items has nothing to do with this patch.

**Comment by DamonFool** [comment]
> The test FAIL and ERROR items has nothing to do with this patch.

Good.
I will run tier1~3 and do a double check of the failing tests.

**Comment by sunny868** [comment]
> > The test FAIL and ERROR items has nothing to do with this patch.
> 
> Good. I will run tier1~3 and do a double check of the failing tests.

Thanks.

**Comment by DamonFool** [comment]
> > The test FAIL and ERROR items has nothing to do with this patch.
> 
> Good. I will run tier1~3 and do a double check of the failing tests.

I tested tier1~3 on Linux/x64, no regression.
So looks good to me.

**Review by DamonFool — APPROVED** [review]

**Review by neliasso — APPROVED** [review]
Looks good. Thanks for fixing!

**Pull request closed without merging**